shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc

```diff
@@ -0,0 +1,297 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/fuchsia/flutter/accessibility_bridge.h"
+
+#include <gtest/gtest.h>
+#include <lib/async-loop/cpp/loop.h>
+#include <lib/async-loop/default.h>
+#include <lib/fidl/cpp/binding_set.h>
+#include <lib/fidl/cpp/interface_request.h>
+#include <lib/sys/cpp/testing/service_directory_provider.h>
+#include <zircon/types.h>
+
+#include <memory>
+
+#include "flutter/lib/ui/semantics/semantics_node.h"
+#include "flutter/shell/platform/fuchsia/flutter/flutter_runner_fakes.h"
+
+namespace flutter_runner_test {
+
+class AccessibilityBridgeTestDelegate
+    : public flutter_runner::AccessibilityBridge::Delegate {
+ public:
+  void SetSemanticsEnabled(bool enabled) override { enabled_ = enabled; }
+  bool enabled() { return enabled_; }
+
+ private:
+  bool enabled_;
+};
+
+class AccessibilityBridgeTest : public testing::Test {
+ public:
+  AccessibilityBridgeTest()
+      : loop_(&kAsyncLoopConfigAttachToCurrentThread),
+        services_provider_(loop_.dispatcher()) {
+    services_provider_.AddService(
+        semantics_manager_.GetHandler(loop_.dispatcher()),
+        SemanticsManager::Name_);
+  }
+
+  void RunLoopUntilIdle() {
+    loop_.RunUntilIdle();
+    loop_.ResetQuit();
+  }
+
+ protected:
+  void SetUp() override {
+    zx::eventpair a, b;
+    zx::eventpair::create(/* flags */ 0u, &a, &b);
+    auto view_ref = fuchsia::ui::views::ViewRef({
+        .reference = std::move(a),
+    });
+
+    accessibility_bridge_ =
+        std::make_unique<flutter_runner::AccessibilityBridge>(
+            accessibility_delegate_, services_provider_.service_directory(),
+            std::move(view_ref));
+    RunLoopUntilIdle();
+  }
+
+  void TearDown() override { semantics_manager_.ResetTree(); }
+
+  MockSemanticsManager semantics_manager_;
+  AccessibilityBridgeTestDelegate accessibility_delegate_;
+  std::unique_ptr<flutter_runner::AccessibilityBridge> accessibility_bridge_;
+
+ private:
+  async::Loop loop_;
+  sys::testing::ServiceDirectoryProvider services_provider_;
+};
+
+TEST_F(AccessibilityBridgeTest, RegistersViewRef) {
+  EXPECT_TRUE(semantics_manager_.RegisterViewCalled());
```

Comment:
These function name should change to RegisterViewForSemantics() ?

Reply:
This is a method on the fake to check whether `RegisterViewForSemantics` was called or not.